groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovycWrapper.java

```diff
@@ -116,6 +131,61 @@ public Path getWorkspaceRoot() {
     }
 
     @Override
+    public Map<String, Set<SymbolInformation>> getReferences() {
+        return references;
+    }
+
+    @Override
+    public Set<Location> findReferences(TextDocumentPositionParams params) {
+        Set<SymbolInformation> symbols = fileSymbols.get(params.getTextDocument().getUri());
+        if (symbols == null) {
+            return Sets.newHashSet();
+        }
+        Set<SymbolInformation> foundSymbolInformations = symbols.stream()
+                .filter(s -> Ranges.isValid(s.getLocation().getRange())
+                        && Ranges.contains(s.getLocation().getRange(), params.getPosition())
+                        && (s.getKind() == SymbolKind.Class || s.getKind() == SymbolKind.Interface))
+                .collect(Collectors.toSet());
+
+        if (foundSymbolInformations.isEmpty()) {
+            return Sets.newHashSet();
+        }
+
+        Optional<SymbolInformation> foundSymbol;
+        if (foundSymbolInformations.size() == 1) {
```

Comment:
I think you can call `reduceFoundSymbolsToMostSpecific` even for a set of size 1? in general this code is very if/else-y which makes it concerning in terms of testability.